plugins/node/opentelemetry-hapi-instrumentation/src/utils.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Attributes } from '@opentelemetry/api';
+import { HttpAttribute } from '@opentelemetry/semantic-conventions';
+import type * as Hapi from '@hapi/hapi';
+import {
+  AttributeNames,
+  HapiLayerType,
+  HapiLifecycleMethodNames,
+} from './types';
+
+export function getPluginName<T>(plugin: Hapi.Plugin<T>): string {
+  if ((plugin as Hapi.PluginNameVersion).name) {
+    return (plugin as Hapi.PluginNameVersion).name;
+  } else {
+    return (plugin as Hapi.PluginPackage).pkg.name;
```

Comment:
I believe there should be no mention of `plugin` now, should be `instrumentation` right ?

Reply:
This is actually referring to the Hapi.Plugin object - confusing overlap in naming schemas, I know!
I will scan over the rest of my code to make sure it is not present elsewhere